File: java/Problem16.java
/*
 * 2^15 = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.
 *
 * What is the sum of the digits of the number 2^1000?
 */

import java.math.BigInteger;

public class Problem16 {
    public static void main(String[] args) {
        int answer = 0;
        BigInteger[] divmod;
        for (BigInteger x = BigInteger.valueOf(2).pow(1000); x.signum() > 0; x = divmod[0]) {
            divmod = x.divideAndRemainder(BigInteger.TEN);
            answer += divmod[1].intValue();
        }
        System.out.println(answer);
    }
}

File: go/Problem16.go
/*
 * 2^15 = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.
 *
 * What is the sum of the digits of the number 2^1000?
 */
package main

import (
	"fmt"
	"math/big"
)

func main() {
	x := big.NewInt(2)
	x.Exp(x, big.NewInt(1000), nil)
	sum := 0
	for _, value := range x.String() {
		sum += int(value) - 48
	}
	fmt.Println(sum)
}
